### PR TITLE
Display unsupported snses in all-voted banner

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

In cases where the user has voted on all available actionable proposals, we currently display the `ActionableProposalsEmpty` banner with the text "You’re all caught up." However, if some SNSes still don’t support actionable proposals, we cannot be sure that there are no votable proposals for these SNSes. To make this clear, the `ActionableProposalsEmpty` banner should display more detailed information. In this PR, we update the description to display the SNSes that may contain votable proposals.

# Changes

- New `joinWithOr` util.
- Display unsupported Snses in `ActionableProposalsEmpty` banner description.
- Not related refactoring: removed redundant array creation [here](https://github.com/dfinity/nns-dapp/pull/4960/files#diff-bdabfefb322354658061b1e40e4e4b5a1fc2feab676ffcd43e37d91b9b7d4894R12).

# Tests

- Test for `joinWithOr`.
- Update `PageBannerPo`.`under` arguments to support it's creation from the root component.
- Test for `ActionableProposalsEmpty` component (now it makes sense, since it includes a dynamic text).

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/tests/page-objects/PageBanner.page-object.ts [hunk 1]
```diff
@@ -9,9 +9,9 @@ export class PageBannerPo extends BasePageObject {
     testId,
   }: {
     element: PageObjectElement;
-    testId: string;
+    testId?: string;
   }): PageBannerPo {
-    return new PageBannerPo(element.byTestId(testId));
+    return new PageBannerPo(testId ? element.byTestId(testId) : element);
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/PageBanner.page-object.ts:14]: What's the reason for this change?
The way you're using it in the test is not correct because the `container` returned by `render` is not itself the `PageBanner` root element.
But why can't you just pass the `testId` from the test?

Comment (mstrasinskis) [reply frontend/src/tests/page-objects/PageBanner.page-object.ts:14]: I think it wouldn’t work because it will try to select an element from the parent element, but the data-tid is placed on the parent element itself.

Comment (mstrasinskis) [reply frontend/src/tests/page-objects/PageBanner.page-object.ts]: Hmm... It didn't work last time I tried, but it is. Maybe wrong id was used. Thanks for noticing.

Comment (mstrasinskis) [reply frontend/src/tests/page-objects/PageBanner.page-object.ts]: Updated https://github.com/dfinity/nns-dapp/pull/4960/files#diff-cb28cc591a2c360ba3e53e3a9b164d562a21f55c6590b2de5bc6fde53ad5726eR36

File: frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    const snsProjects = Array.from(
+      new Array(includeBallotsByCallerList.length)
+    ).map((_, i) => ({
+      lifecycle: SnsSwapLifecycle.Committed,
+      rootCanisterId: principal(i),
+      projectName: `SNS-${i}`,
+    }));
+    setSnsProjects(snsProjects);
+
+    Array.from(new Array(includeBallotsByCallerList.length)).forEach((_, i) =>
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts:22]: Do you need both `Array.from` and `new Array`? Seem redundant?

Do you even need either of them? Does this work?
```
includeBallotsByCallerList.forEach((includeBallotsByCaller, i) =>
      actionableSnsProposalsStore.set({
        rootCanisterId: principal(i),
        proposals: [],
        includeBallotsByCaller,
      })
    ); 
```

File: frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    return PageBannerPo.under({
+      element: new JestPageObjectElement(container),
+      testId: "actionable-proposals-empty",
+    });
+  };
+
+  beforeEach(() => {
+    resetSnsProjects();
+    actionableSnsProposalsStore.resetForTesting();
+    allowLoggingInOneTestForDebugging();
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts:43]: please remove

File: frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts]: Sure, has been simplified. Thanks!

File: frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/proposals/ActionableProposalsEmpty.spec.ts]: Good catch, removed 🙈
